### PR TITLE
Add resilience4j dependency and implementation

### PR DESCRIPTION
**Pull request by 1jeanpaul1**
## Problem
Need a fault tolerance library in our service

## Solution
Implement resilience4j for fault tolerance. This PR only implements Circuit Breakers but other useful tools can be added 

You can find out more about resilience4j here: https://resilience4j.readme.io/docs/

## Test
1. build and run: `docker-compose up --build`
2. Go to http://localhost:8082/dev/fallback and verify that you get the following response:
`{"code":-1,"httpCode":400,"message":"Fallback: TESTING TESTING TESTING"}`
3. Keep refreshing the page (about 10-12 times), verify that you get this new response:
`{"code":-1,"httpCode":500,"message":"Fallback: CircuitBreaker 'backendA' is OPEN and does not permit further calls"}`

Explanation: the http://localhost:8082/dev/fallback endpoint is always returning an IllegalArgumentException. The fallback for that exception is `{"code":-1,"httpCode":400,"message":"Fallback: TESTING TESTING TESTING"}`. When you keep refreshing the page, the server notices that the same exception is being thrown. So the server now rejects the requests for a set number of seconds. 

This would be useful in cases where our database related queries are taking too long and timing out. It would give our database some breathing room to try to catch itself up.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,3 +35,36 @@ You can access the DB through pgadmin
     * password: password
     * user: postgres
 
+## Patterns
+
+**TODO: Move these to a wiki**
```
Comment (1jeanpaul1) [inline README.md:40]: Will probably want to move these to a wiki eventually

File: src/main/java/com/fcgl/madrid/dev/DataPopulation.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.fcgl.madrid.dev;
 
-import com.fcgl.madrid.forum.model.Comment;
-import com.fcgl.madrid.forum.model.Post;
+import com.fcgl.madrid.forum.dataModel.Comment;
```
Comment (1jeanpaul1) [inline src/main/java/com/fcgl/madrid/dev/DataPopulation.java:3]: Moved all database related models into `dataModel` package.

File: src/main/java/com/fcgl/madrid/forum/model/InternalStatus.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.fcgl.madrid.forum.model;
+
+public class InternalStatus {
```
Comment (1jeanpaul1) [inline src/main/java/com/fcgl/madrid/forum/model/InternalStatus.java:3]: Returning a POJO in your controller returns json to the frontend. This class has pretty much everything we need for error responses. But will probably want to modify or delete altogether depending on our use case

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -25,3 +25,19 @@ ribbon:
   eureka:
     enabled: true
 
+
+#Resilience4j config: Have to run some tests to figure out best config
```
Comment (1jeanpaul1) [inline src/main/resources/application.yml:29]: AS mentioned in the comment, going to want to test what config is best for our system in the future. This is just for now

**Review by Mikerad1 — APPROVED** [review]
Does what is said. dont know how fault tolerance works so will trust

**Comment by fcgl** [comment]
@Mikerad1 we'll see how it goes haha

**Pull request merged**